library/include/modules/Gui.h

```diff
@@ -107,6 +110,10 @@ namespace DFHack
         DFHACK_EXPORT df::building *getAnyBuilding(df::viewscreen *top);
         DFHACK_EXPORT df::building *getSelectedBuilding(color_ostream &out, bool quiet = false);
 
+        DFHACK_EXPORT bool any_stockpile_hotkey(df::viewscreen* top);
+        DFHACK_EXPORT df::building_stockpilest *getAnyStockpile(df::viewscreen* top);
+        DFHACK_EXPORT df::building_stockpilest *getSelectedStockpile(color_ostream& out, bool quiet = false);
```

Comment:
`any_stockpile_hotkey` is good, but I do recommend against `getAnyStockpile` and `getSelectedStockpile` unless there is a proven need for them. This will affect all tools that implement the `getSelected*` API (e.g. `gui/launcher`)

actually, let me walk that back. if we can treat a non-focused ZScreens as invisible, then there is no need for these tools to implement these functions just to pass them through to the lower layer. So go ahead and add these. We can clean up the tools later and simplify them instead of make them more complicated.